src/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -103,9 +103,19 @@ private function setHtml(string $html)
      * Provides access to the internal DOMDocument representation of the HTML in its current state.
      *
      * @return \DOMDocument
+     *
+     * @throws \UnexpectedValueException
      */
     public function getDomDocument(): \DOMDocument
     {
+        if (null === $this->domDocument) {
+            throw new \UnexpectedValueException(
+                self::class .
+                '::setDomDocument() has not yet been called on ' .
+                static::class
+            );
```

Comment:
Could we have an exception code based on the current UNIX timestamp?  (It will also mark your change on the timeline of all history, up until at least 2037, etc. :>)

Reply:
Also I prefer to have the `.` operators on the beginning of the next line rather than end of the preceding.  But I don't think we have any stylistic rule about that - @oliverklee ?

It might become clearer when you add the 2nd argument - is the code easier to read with the `.` operator at the start of a new line?  YMMV, and it's no big deal...

Reply:
> It might become clearer when you add the 2nd argument - is the code easier to read with the `.` operator at the start of a new line? YMMV, and it's no big deal...

my general preference is parenthesis.

Reply:
resolved in 0bf5794 and 0adb26f

Reply:
> my general preference is parenthesis.

Good idea.  LGTM now :)